assets/js/components/user-input/UserInputSelectOptions.js

```diff
@@ -178,17 +186,52 @@ export default function UserInputSelectOptions( {
 		}
 
 		return (
-			<div
-				key={ optionSlug }
-				className="googlesitekit-user-input__select-option"
-			>
-				<ListComponent { ...props }>
-					{ options[ optionSlug ] }
-				</ListComponent>
-			</div>
+			'sell_products_or_service' !== optionSlug && (
+				<div
+					key={ optionSlug }
+					className="googlesitekit-user-input__select-option"
+				>
+					<ListComponent { ...props }>
+						{ options[ optionSlug ] }
+					</ListComponent>
+				</div>
+			)
 		);
 	} );
 
+	const currentlyEditingSlug = useSelect( ( select ) =>
+		select( CORE_UI ).getValue( USER_INPUT_CURRENTLY_EDITING_KEY )
+	);
+
+	const previousValues = usePrevious( values );
+
+	useEffect( () => {
+		if (
+			currentlyEditingSlug === USER_INPUT_QUESTIONS_PURPOSE &&
+			values.includes( 'sell_products_or_service' ) &&
+			'sell_products' in options
+		) {
+			if (
+				undefined !== previousValues &&
+				previousValues.includes( 'sell_products_or_service' )
+			) {
+				setUserInputSetting( slug, [ 'sell_products' ] );
+				setValues( FORM_USER_INPUT_QUESTION_SNAPSHOT, {
+					[ slug ]: [ 'sell_products' ],
+				} );
+			} else {
+				setUserInputSetting( slug, values );
+			}
+		}
+	}, [
+		options,
```

Comment:
You missed `setValues` in the dependency array

Reply:
This has been refactored in the last commit or so @zutigrm